MAINT - Use real enums and parent class members

carlwilson opened the pull request:
- moved repeatedly used `_ckSummer`, `_cstream`, and `_dstream` members to `ModuleBase`;
- tidied `ModuleBase` parameter handling;
- added a common method to set up module checksumming streams `ModuleBase.setupDataStream()`;
- `BytestreamModule` uses `ModuleBase.setupDataStream()`;
- replaced all types that extended `EnumerationType` with real Java `enum` types:
  - `AgentType`,
  - `AnalogDigitalFlagType`,
  - `ChecksumType`,
  - `DocumentType,`
  - `IdentifierType`,
  - `PropertyArity`,
  - `PropertyType`,
  - `SignatureType`,
  - `SignatureUseType`, and
- removed now defunct `EnumerationType`.

Closes #373

carlwilson committed:
MAINT - Use real enums and parent class members
- moved repeatedly used `_ckSummer`, `_cstream`, and `_dstream` members to `ModuleBase`;
- tidied `ModuleBase` parameter handling;
- added a common method to set up module checksumming streams `ModuleBase.setupDataStream()`;
- `BytestreamModule` uses `ModuleBase.setupDataStream()`;
- replaced all types that extended `EnumerationType` with real Java `enum` types:
  - `AgentType`,
  - `AnalogDigitalFlagType`,
  - `ChecksumType`,
  - `DocumentType,`
  - `IdentifierType`,
  - `PropertyArity`,
  - `PropertyType`,
  - 'SignatureType',
  - `SignatureUseType`, and
- removed now defunct `EnumerationType`.

carlwilson committed:
FIX - SHA1 name output typo.

carlwilson committed:
Merge branch 'integration' into refact/module-base

david-russo commented on jhove-core/src/main/java/edu/harvard/hul/ois/jhove/IdentifierType.java:
```diff
@@ -5,111 +5,113 @@
-     * CLASS CONSTRUCTOR.
-     ******************************************************************/
+	/**
+	 * Applications will never create SignatureTypes directly.
```
Incorrect `*Type` referenced in comment. An old copy-paste error, though we probably don't need any of these comments anymore anyway since all `enum` constructors must be `private`. I'd do away with the large "CLASS CONSTRUCTOR"-style sign-boards as well, at least for simple enums.

david-russo commented on jhove-core/src/main/java/edu/harvard/hul/ois/jhove/module/BytestreamModule.java:
```diff
@@ -96,18 +87,7 @@ public final int parse (InputStream stream, RepInfo info, int parseIndex)
-            _dstream = getBufferedDataStream (stream, _je != null ?
-                                _je.getBufferSize () : 0);
-        }
+        setupDataStream(stream, info);
```
The comment preceding this call explains code now hidden inside `setupDataStream`, so it would make more sense there instead.

david-russo commented on jhove-core/src/main/java/edu/harvard/hul/ois/jhove/module/BytestreamModule.java:
```diff
@@ -129,14 +109,14 @@ public final int parse (InputStream stream, RepInfo info, int parseIndex)
 	}
-        if (ckSummer != null) {
-	    info.setChecksum (new Checksum (ckSummer.getCRC32 (), 
+        if (_ckSummer != null) {
```
I was going to suggest moving this checksum-related code to `ModuleBase` as well, but apparently there's already a `setChecksums` method there which you could replace this with. Might want to keep an eye out in other modules as well.

carlwilson replied:
I'm on the trail of a few Module similarities. The calculation of checksums is one, the proffering and checking of sigs is another. All of these were supposed to be abstract class tasks but cut and paste has prevailed.

david-russo reviewed (commented):
Really good stuff :)

carlwilson committed:
REVIEW - Addressing review comments
- picked up a few more variable first `var.equals(const)` constructs;
- improved and cleaned up comments in enumerated types;
- tidied a few nested `if else` constructs;
- `BytestreamModule` now uses `ModuleBase.setChecksums()` method; and
- begun to add log statements for ingored exceptions.
